src/app/shared/pipes/highlight-matching-text-pipe.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * This pipe takes a string and a substring to search, and returns an array of {text: string, highlighted: boolean}
+ * @copyright The GHGA Authors
+ * @license Apache-2.0
+ */
+
+import { Pipe, PipeTransform } from '@angular/core';
+
+/**
+ * This pipe returns an array of {text: string, highlighted: boolean} based on a given string
+ * and a substring to search for in the given string. Those parts of the given string that match
+ * the substring, we highlight.
+ */
+@Pipe({
+  name: 'HighlightMatchingTextPipe',
+})
+export class HighlightMatchingText implements PipeTransform {
```

Comment:
I would make this a utility function in metadata/utils or shared/utils, since it is not used as a pipe, but as a helper function in computed signals.